cdc/log.cc

```diff
@@ -526,36 +551,173 @@ class transformer final {
                     r.for_each_cell([&](column_id id, const atomic_cell_or_collection& cell) {
                         auto& cdef = _schema->column_at(ckind, id);
                         auto* dst = _log_schema->get_column_definition(to_bytes("_" + cdef.name()));
-                        // TODO: collections.
-                        if (cdef.is_atomic()) {
-                            column_op op;
+                        auto res_type = static_pointer_cast<const tuple_type_impl>(dst->type)->type(1);
```

Comment:
I don't understand what this res_type is, or what type(1) does (sorry...) but looking below, it seems it is only used in the non-atomic case? So maybe it can be moved closer to where it's needed?

Reply:
It is the type of the log value (second element of the column tuple). And, yes, it could be declared lower down now.

Reply:
Moved to local usage points.